src/Cli/SignalLoop.php

```diff
@@ -44,9 +43,29 @@ class SignalLoop implements LoopInterface
      * @var bool status when suspend or resume signal was got.
      */
     protected bool $pause = false;
+    private int $memorySoftLimit;
+    private EventDispatcherInterface $dispatcher;
+
+    /**
+     * @param EventDispatcherInterface $dispatcher
+     * @param int $memorySoftLimit Soft RAM limit in bytes. The loop won't let you continue to execute the program if soft limit is reached. Zero means no limit.
+     * @param int[] $exitSignals pcntl signal codes to exit the loop
+     * @param int[] $suspendSignals pcntl signal codes to pause loop execution
+     * @param int[] $resumeSignals pcntl signal codes to resume loop execution
+     */
+    public function __construct(
+        EventDispatcherInterface $dispatcher,
+        $memorySoftLimit = 0,
+        $exitSignals = [1, 2, 15], // SIGHUP, SIGINT, SIGTERM
```

Comment:
is it better to use constants?

Reply:
Constants are not available when pcntl module is not loaded. Pcntl module is optional for this package.